File: top-interview-questions-easy/single-number.go
func singleNumber(nums []int) int {
    result := 0
    sort.Ints(nums)
    
    dic := map[int]int{}
    
    for _, v := range nums {
        dic[v]++
    }
    

    for k, v := range dic {
        if v == 1 {
            result = k
        }
    }
    
    return result
}

File: top-interview-questions-easy-2/single-number.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        dic = dict()
        
        for i in nums:
            dic[i] = dic.get(i, 0) + 1
            
        for num in dic:
            if dic[num] == 1:
                return num

